hippoclient/textedit.py

```diff
@@ -0,0 +1,462 @@
+        ("m", "show_tab('metadata-tab')", "Metadata"),
+    ]
+
+    def __init__(self, id: PydanticObjectId, client: Client):
+        self._object_id = id
+        self._client = client
+
+        _product = product.read(client, id)
+
+        self._current_version = _product.version
+        self._projected_version = self.project_version()
+
+        super().__init__()
+
+        self._product_name = _product.name
+        self._description = _product.description
+        self._sources = _product.sources
+        self._metadata = _product.metadata
+
+    def compose(self) -> ComposeResult:
+        # """Create child widgets for the app."""
+        yield Footer()
+
+        with TabbedContent():
+            with TabPane("Details", id="details-tab"):
+                with Vertical(classes="title"):
+                    yield Label("Product Title:", variant="accent")
+                    yield Input(
+                        value=self._product_name,
+                        placeholder="Product title...",
+                        id="product-title",
+                    )
+                with Vertical(classes="description"):
+                    yield Label("Product Description:", variant="accent")
+                    yield TextArea(
+                        classes="textarea",
+                        text=self._description,
+                        language="markdown",
+                        id="product-description",
+                    )
+                with Vertical(classes="versioning"):
+                    yield Label("Version Bump:", variant="accent")
+                    yield RadioSet(
+                        RadioButton("Major"),
+                        RadioButton("Minor"),
+                        RadioButton("Patch", value=True),
+                        id="version",
+                    )
+                    yield Label(
+                        f"{self._current_version} → {self._projected_version}",
+                        id="versionproject",
+                        variant="success",
+                    )
+                with Horizontal(classes="save-or-cancel"):
+                    yield Button("Save", variant="success", id="save")
+                    yield Button("Cancel", variant="error", id="cancel")
+
+            with TabPane("Sources", id="sources-tab"):
+                for source in self._sources:
+                    with Horizontal(classes="sources existing-source"):
+                        with Vertical():
+                            yield Label("Filename:", variant="accent")
+                            yield Input(
+                                value=source.name,
+                                placeholder="",
+                                id=f"edit-source-name-{source.uuid}",
+                                tooltip="Enter the absolute filepath to a file of the same name to update the file; leave this field as-is if only editing other fields (i.e., source description)",
```

Comment:
Can you break up this line?

(
  "Enter the absol..."
  "update the file..."
  "other fields..."
)